### ISSUE TITLE
Azure CLI in Docker not working

### ISSUE THREAD

**Issue by mkoziel2000** [Azure/azure-cli#16341]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
Whenever running a windows container on Docker in windows 2019 as the host, the "az" CLI command immediately returns rather than showing the banner and help information.  The problem looks to be with the "python.exe" that it is calling in the CLI2 directory.  That program seems to immediately return with no output when called from the commandline.  

It seems to be related to the 12-2020 Cumulative Update (or some update since May 2020).  When I uninstall the Cumulative update (machine reverts back to the may update), then it all starts working again...Python.exe returns an interactive session and the "az" commands work.

**To Reproduce**
Install Docker on a Win2019 server (1809).  Run a container that has the latest Azure CLI tools on it 2.16.0.  Apply 12-2020 Cumulative update.  Interact with the container via "docker exec" and try to run "az"...the behavior is that it immediately returns without displaying anything.

**Expected behavior**
Display the commandline options when "az" is run.

**Environment summary**
Azure tools was installed into docker image using the following commands in the Dockerfile:

 RUN choco install az.powershell -y --version 4.4.0
 RUN Invoke-WebRequest -Uri https://azurecliprod.blob.core.windows.net/msi/azure-cli-2.9.1.msi -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; del AzureCLI.msi

**Additional context**
Running on Docker 19.03
Windows Server 2019 (build 17763)

**Comment by mkoziel2000** [comment]
FYI...Loading the most recent version of the Azure CLI tools into the image (2.16.0) didn't seem to improve anything.

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls have a look? thanks

**Comment by mkoziel2000** [comment]
Any progress on the diagnosis?  I currently have our production servers patching suspended in the hopes we can figure out what might be at fault here.  Is this something that needs to be pursued with the python folks in parallel (whoever that might be)?  Are others having this issue?

**Comment by fengzhou-msft** [comment]
@mkoziel2000 it looks like a shell invoke issue with the python.exe caused by the Cumulative update. Can you check the results for the below commands in powershell:
`& "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az.cmd"`
`& "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe" -IBm azure.cli`

**Comment by mkoziel2000** [comment]
The first command just returns...no output.  That was part of the original testing.  The second test was done without the command tail and that also just returned without any output,  I'll try it with the command tail to see if it produces anything.

**Comment by MathieuHamel** [comment]
I got the same problem and finally came to conclusion that it could be related to [this](https://support.microsoft.com/en-us/topic/you-might-encounter-issues-when-using-windows-server-containers-with-the-february-11-2020-security-update-release-b9a8fcae-950d-7a0b-ac7c-cb6b294cb809) the Windows container version compatibility. 

I was able to fix this by updating my container host with the latest patches and in conjunction with latest Windows Server core images.

I hope it can help.

**Comment by fengzhou-msft** [comment]
@mkoziel2000 can you follow @MathieuHamel's comment and give it a try?

**Comment by yungezz** [comment]
hi @mkoziel2000 could you pls try the solution mentioned above? Will close the issue for now since it's container issue.  Feel free to create new issue if any other error. thanks.

**Comment by mkoziel2000** [comment]
It seems to be working with April 2021 CU on Windows 2019.
